source/nn/positional-encoding.lisp

```diff
@@ -1,6 +1,39 @@
 (in-package :caten/nn)
 
-;; from https://github.com/ml-explore/mlx/blob/main/python/mlx/nn/layers/positional_encoding.py
+; from https://github.com/ml-explore/mlx/blob/main/python/mlx/nn/layers/positional_encoding.py
 ;; RoPE
 ;; PositionalEncoding
 
+(defmodel (RoPE (dims &key (traditional NIL) (base 10000) (scale 1.0)))
+  ((dims dims)
+   (traditional traditional)
+   (base base)
+   (scale scale)))
+
+;RoPE needs an extra arg in the call function, the generic doesn't consider it
+;offset: int = 0
+(defmethod call ((op RoPE) &rest inputs)
```

Comment:
`:offset` should passed as an argument of RoPE model, not an argument for `call`